dace/frontend/common/distr.py

```diff
@@ -16,17 +16,159 @@
+
+    # Dummy tasklet adds MPI variables to the program's state.
+    from dace.libraries.mpi import Dummy
+    tasklet = Dummy(pgrid_name, [
+        f'MPI_Comm {pgrid_name}_comm;',
+        f'MPI_Group {pgrid_name}_group;',
+        f'int {pgrid_name}_coords[{pgrid_ndims}];',
+        f'int {pgrid_name}_dims[{pgrid_ndims}];',
+        f'int {pgrid_name}_rank;',
+        f'int {pgrid_name}_size;',
+        f'bool {pgrid_name}_valid;',
+    ])
+
+    state.add_node(tasklet)
+
+    # Pseudo-writing to a dummy variable to avoid removal of Dummy node by transformations.
+    _, scal = sdfg.add_scalar(pgrid_name, dace.int32, transient=True)
+    wnode = state.add_write(pgrid_name)
+    state.add_edge(tasklet, '__out', wnode, None, Memlet.from_array(pgrid_name, scal))
+
+    return pgrid_name
+
+
+@oprepo.replaces_method('ProcessGrid', 'Sub')
+def _pgrid_sub(pv: 'ProgramVisitor', sdfg: SDFG, state: SDFGState, parent_grid: str, color: Sequence[Union[Integral,
+                                                                                                           bool]]):
+    """ Equivalent to `dace.comm.Cart_sub(parent_grid, color).
+        :param parent_grid: Parent process-grid (similar to the `comm` parameter of `MPI_Cart_sub`).
+        :param color: The i-th entry specifies whether the i-th dimension is kept in the sub-grid or is dropped (see `remain_dims` input of `MPI_Cart_sub`).
+        :return: Name of the new sub-grid descriptor.
+    """
+
+    return _cart_sub(pv, sdfg, state, parent_grid, color)
+
+
+@oprepo.replaces_operator('ProcessGrid', 'Eq', otherclass='Comm')
+@oprepo.replaces_operator('ProcessGrid', 'Is', otherclass='Comm')
+def _pgrid_eq_comm(pv: 'ProgramVisitor', sdfg: SDFG, state: SDFGState, op1: str, op2: 'Comm'):
+    from mpi4py import MPI
+    if op2 is MPI.COMM_WORLD or op2 is MPI.COMM_NULL:
+        return False
+    return True
+
+
+@oprepo.replaces_operator('Comm', 'Eq', otherclass='ProcessGrid')
+@oprepo.replaces_operator('Comm', 'Is', otherclass='ProcessGrid')
+def _comm_eq_pgrid(pv: 'ProgramVisitor', sdfg: SDFG, state: SDFGState, op1: 'Comm', op2: 'str'):
+    from mpi4py import MPI
+    if op1 is MPI.COMM_WORLD or op1 is MPI.COMM_NULL:
+        return False
+    return True
+
+
+@oprepo.replaces_operator('ProcessGrid', 'NotEq', otherclass='Comm')
+@oprepo.replaces_operator('ProcessGrid', 'IsNot', otherclass='Comm')
+def _pgrid_neq_comm(pv: 'ProgramVisitor', sdfg: SDFG, state: SDFGState, op1: str, op2: 'Comm'):
+    from mpi4py import MPI
+    if op2 is MPI.COMM_WORLD or op2 is MPI.COMM_NULL:
+        return True
+    return False
+
+
+@oprepo.replaces_operator('Comm', 'NotEq', otherclass='ProcessGrid')
+@oprepo.replaces_operator('Comm', 'IsNot', otherclass='ProcessGrid')
+def _comm_neq_pgrid(pv: 'ProgramVisitor', sdfg: SDFG, state: SDFGState, op1: 'Comm', op2: 'str'):
+    from mpi4py import MPI
+    if op1 is MPI.COMM_WORLD or op1 is MPI.COMM_NULL:
+        return True
+    return False
```

Comment:
Do these do anything?

Reply:
Updated. Needed for communicator comparisons, but will be revisited in future PRs.